tests/testthat/test-as_curve_params.R

```diff
@@ -0,0 +1,20 @@
+test_that("`as_curve_params()` produces an error when non-curve data is provided", {
+  library(magrittr)
+  expect_error(
+    object = curve_data <-
+      "https://osf.io/download//n6cp3/" %>% # pop data
+      readr::read_rds() %>%
+      as_curve_params(),
+    regexp = "Please provide curve data",
```

Comment:
this is a good opportunity to upgrade from using `regexp` to using `class` to test error conditions; see explanation [here](https://testthat.r-lib.org/reference/expect_error.html?q=class#testing-message-vs-class)  and example [here](https://r-pkgs.org/testing-basics.html#:~:text=Recent%20developments%20in%20both%20base%20R%20and%20rlang%20make%20it%20increasingly%20likely%20that%20conditions%20are%20signaled%20with%20a%20class%2C%20which%20provides%20a%20better%20basis%20for%20creating%20precise%20expectations.%20That%20is%20exactly%20what%20you%E2%80%99ve%20already%20seen%20in%20this%20stringr%20example.%20This%20is%20what%20the%20class%20argument%20is%20for%3A)

Reply:
I have made the switch from `regexp` to `class` to capture the error.